modules/auxiliary/gather/ldap_query.rb

```diff
@@ -184,6 +184,101 @@ def generate_rex_tables(entries, format)
     end
   end
 
+  # Decode a binary string containing a GUID into a GUID string
+  # of the format XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX
+  def decode_guid_binstring(guid)
+    # See https://docs.microsoft.com/en-us/dotnet/api/system.guid.tobytearray?view=netcore-3.1#remarks
+    # for more information. Specifically note that they state that the first four byte group and the
+    # following 2 groups of two bytes each are reversed. Following this, the next 2 byte group and
+    # the terminating 4 byte group are kept in the same order as in the binary data.
+    #
+    # Thanks to Spencer McIntyre for providing assistance on this code.
+    decoded_guid = [guid[0...4].reverse, guid[4...6].reverse, guid[6...8].reverse, guid[8...10], guid[10..]].map { |p| p.unpack1('H*') }.join('-')
+    decoded_guid = decoded_guid.upcase # All GUID strings must be uppercase
+
+    decoded_guid
```

Comment:
Should probably update this to using BinData in some way if we can vs custom unpacks.

Reply:
Hmm so at the moment we have a definition over at https://github.com/rapid7/ruby_smb/blob/338d2a6f5c2daa62c767e22cb59086e121853d72/lib/ruby_smb/dcerpc/uuid.rb that we were using for GUID's however the problem I'm seeing there is that we don't have a way to go from a raw binary string to a BinData version of the GUID string. 

The only option presently is to go from the converted GUID string version to a BinData representation of that string, not from the raw binary itself.